### ISSUE TITLE
Comments removed when formatting

### ISSUE THREAD

**Issue by krazik** [adamvoss/vscode-yaml#9]
I can't figure out how to stop it but this plugin removes all comments `# Foo` from my yaml files.

**Comment by adamvoss** [comment]
I recognize this is not an ideal solution, but the solution is to not use the code formatting functionality.

The underlying library that provides formatting [does not (and will not) support comments](https://github.com/nodeca/js-yaml/issues/196).  If another workable library can be found that preserves comments, I would be happy to switch.

**Comment by krazik** [comment]
yeah that sucks I obviously understand too.  thanks for the reply.

-Ry

**Comment by adamvoss** [comment]
Reopening as a placeholder to try to reduce issue duplication.

**Comment by txomon** [comment]
Would you mind listing this in `Known Issues` section? Thanks!

**Comment by adamvoss** [comment]
@txomon That is a good idea.  It will require a new release of `vscode-yaml` in order to show on the marketplace and I'm currently working on some potentially big changes regarding this extension, so I might let some of that hash out in the next couple days first.

**Comment by ecstasy2** [comment]
+1
This is a deal breaker for me. Yaml is code and code need documentation. It would be nice if we can find a alternative.

**Comment by fullc0ntr0l** [comment]
https://github.com/mohsen1/yawn-yaml

**Comment by BYK** [comment]
@adamvoss is there any possibility to replace the parser witn `yawn-yaml` as @fullc0ntr0l suggested? I can open a PR if you are okay with the idea.
